src/test-run/request/create-request-options.ts

```diff
@@ -173,16 +169,31 @@ function getProxyUrl (testRun: TestRun, url: string, withCredentials?: boolean):
     }, testRun, true)) as Promise<string>;
 }
 
+async function resolveUrlParts (testRun: TestRun, url: URL, withCredentials: boolean): Promise<{ hostname: string; port: string; href: string; partAfterHost: string }> {
+    if (testRun.isProxyless()) {
+        const { href, hostname, port, pathname, search } = url;
+        const partAfterHost      = [pathname, search].join('');
+
+        return { partAfterHost, href, hostname, port };
+    }
+
+    const href               = await getProxyUrl(testRun, url.href, withCredentials);
+    const urlObj             = await parseProxyUrl(href);
+    const { partAfterHost }  = urlObj;
+    const { hostname, port } = urlObj.proxy;
+
+    return { partAfterHost, href, hostname, port };
+}
+
 export async function createRequestOptions (currentPageUrl: URL, testRun: TestRun, options: ExternalRequestOptions, callsite: CallsiteRecord | null): Promise<RequestOptions> {
     options.headers = options.headers || {};
 
-    const url             = await prepareUrl(testRun, currentPageUrl, options.url, callsite);
-    const withCredentials = !currentPageUrl.host || sameOriginCheck(currentPageUrl.href, url.href) || options.withCredentials || false;
-    const body            = transformBody(options.headers, options.body);
-    const headers         = await prepareHeaders(options.headers, currentPageUrl, url, body, testRun, withCredentials, options);
-    const proxyUrl        = await getProxyUrl(testRun, url.href, withCredentials);
-    const proxyUrlObj     = parseProxyUrl(proxyUrl);
-    let auth              = options.auth;
+    const url                                     = await prepareUrl(testRun, currentPageUrl, options.url, callsite);
+    const withCredentials                         = !currentPageUrl.host || sameOriginCheck(currentPageUrl.href, url.href) || options.withCredentials || false;
+    const body                                    = transformBody(options.headers, options.body);
+    const headers                                 = await prepareHeaders(options.headers, currentPageUrl, url, body, testRun, withCredentials, options);
+    const { hostname, port, href, partAfterHost } = await resolveUrlParts(testRun, url, withCredentials);
```

Comment:
Destructurize these variables into different lines to reduce spaces before `=` in other assignments.